### DISCUSSION TITLE
[tooltip] add support for native popover API

### DISCUSSION THREAD

**Discussion by brianferry** [RedHat-UX/red-hat-design-system#1984] [Ideas]
`rh-tooltip` currently uses `floating-ui` in order to handle its tooltip implementation.  

In the effort of loading the least amount of JavaScript and the maximum about of browser support we should consider adding a version of the tooltip that uses the [Popover API](https://developer.mozilla.org/en-US/docs/Web/API/Popover_API).  

The reasoning for adding a version of the tooltip instead of rewriting the entire tooltip currently is **browser support**.  

Currently, many of the popover features are available in `Chrome` but not in `Safari` or `Firefox`.  We could add a check for browser support and if available load the native HTML element version of the popover and have the floating-ui as a fallback for Safari and Firefox.  

Additional considerations:

- Accessibility of the popover element
- Having to recreate the fallback logic in JavaScript
  - If the user positions the tooltip to the left but it does not have room to show we need to move the tooltip where it will be visible.
- Differences between the naming scheme of floating-ui and the native HTML element.  Some positions in floating-ui are not duplicated in the native element and so would either need to be created or deprecated. 
- Styling considerations
  - Are we able to create something like the caret on the tooltip with the native API?

**Comment by bennypowers** [comment]
Correct me if I'm wrong, but we'll also need the [anchor positioning](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_anchor_positioning) css proposal in order to remove floating-ui?
